pennylane/qnode.py

```diff
@@ -717,6 +722,9 @@ def __call__(self, *args, **kwargs):  # pylint: disable=too-many-branches
             # last axis corresponding to the number of qubits
             return qml.math.squeeze(res, axis=0)
 
+        if old_interface == "auto":
+            self.interface = "auto"

```

Comment:
Should this be moved further up to line 714 or so? Seems like this is not being executed in all cases because the method may have returned a value before already.

Reply:
Good catch! I moved this code right after computing the results, and also added it inside the new returns type branch.